libnetwork/endpoint.go

```diff
@@ -96,6 +96,18 @@ type Endpoint struct {
 	mu                sync.Mutex
 }
 
+type optionDataType int
+
+const (
+	optionDataSandbox optionDataType = iota + 1
+	optionDataJoinOpts
+	optionDataAddLegacyLinks
+)
+
+// optionData is used by functional options to store data that does not belong
+// in the Endpoint itself.
+type optionData map[optionDataType]any
```

Comment:
Yes, I started with a struct but because the same `EndpointOption` type is used for both create and join options, it meant having struct members that were only of use to one or the other. So, I tried this.

I don't like it either though, I'll go back to a struct.